pkg/crc/config/validations.go

```diff
@@ -123,7 +123,7 @@ func ValidateYesNo(value interface{}) (bool, string) {
 func validatePreset(value interface{}) (bool, string) {
 	_, err := crcpreset.ParsePresetE(cast.ToString(value))
 	if err != nil {
-		return false, fmt.Sprintf("Unknown preset. Only %s and %s are valid.", crcpreset.Podman, crcpreset.OpenShift)
+		return false, fmt.Sprintf("Unknown preset. Only %s, %s and %s are valid.", crcpreset.Podman, crcpreset.OpenShift, crcpreset.Microshift)
```

Comment:
I think I'd add a `preset: Add Microshift preset value` commit doing all the `case Microshift`, `if preset.OpenShift`, ... changes to separate these almost mechanical changes, but happening in multiple files , from the most important part of the PR, which is the changes in `pkg/machine/start.go`

You can still error out in `pkg/crc/config/validations.go` if trying to use the `microshift` preset so that it's not possible to try to use this preset before the commit with the `start.go` changes.

Reply:
Just to have different commit without any functionality doesn't make sense if I put a different commit for preset specific changes and then error our as part of validation then again it is  dead code for that commit.

Reply:
Yes, these are a lot of small/trivial changes spread over a lot of files. If you move out to their own commit, this commit is quick to review, and the 'main commit' is smaller/more focused, which means easier to review.